src/background.js

```diff
@@ -4,9 +4,66 @@ import 'src/activity-logger/background'
 import 'src/omnibar'
 import { installTimeStorageKey } from 'src/imports/background'
 import convertOldData from 'src/util/old-data-converter'
+import fetchPageData from 'src/page-analysis/background/fetch-page-data'
+import { findPagesByUrl } from 'src/search/find-pages'
+import { transformToBookmarkDoc } from 'src/imports/background/imports-preparation'
+import { dataURLToBlob } from 'blob-util'
+import { generatePageDocId } from 'src/page-storage'
+import db from 'src/pouchdb'
 
 export const dataConvertTimeKey = 'data-conversion-timestamp'
 
+
+export async function bookmarkStorageListener(id, bookmarkInfo) {
+    const urlToFind = bookmarkInfo.url
+    const samePageCandidates = (await findPagesByUrl({urlToFind})).rows.map(row => row.doc)
+    if (samePageCandidates.length > 0) {
+        let pageDoc = {}
+        for (var i = samePageCandidates.length - 1; i >= 0; i--) {
```

Comment:
Iterating over the results isn't necessary here. If you look at the code for `findPagesByUrl` it uses exact match of the `url` in the query selector, hence this loop is always going to `break` on the first value. Basically you should only need the last page doc it finds. They should be sorted by time, but not sure which order. Either the first or last result should be the one you want to assign `pageDoc` to